### ISSUE TITLE
Using --verifytypes on a library with a declared variable doesn't infer type

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#1608]
**Describe the bug**
This is from pandas.  We have a variable assigned to an instance of a class within a module.  With `--verifytypes`, pyright says that the variable is missing a type annotation, but I don't believe it should be needed.

**To Reproduce**
Create a folder `ctlib` .  Add a blank `__init__.py` and blank `py.typed` file to it.  Then create `classtypes.py` in `ctlib` containing:
```python
# This is classtypes.py
from typing import List


class Foo:
    """Foo
    """

    def __init__(self):
        pass

    def __repr__(self) -> str:
        return "A Foo"


def add_a_foo(f: Foo) -> None:
    """add_a_foo
    """
    registry.register(f)


class Registry:
    """Registry
    """

    def __init__(self):
        self._foolist: List[Foo] = []

    def register(self, f: Foo) -> None:
        """register
        """
        self._foolist.append(f)

    def __repr__(self) -> str:
        return self._foolist.__repr__()

registry = Registry()

add_a_foo(Foo())
add_a_foo(Foo())
print(registry)


```
If you do `pyright --verifytypes ctlib`, you get
```
 ctlib\classtype.py:37:1 - error: Type not declared for variable "ctlib.classtype.registry"
```
If you do `pyright ctlib/classtype.py`, no error is reported.

**Expected behavior**
No error when the variable is not typed and using `--verifytypes`.  Type can be inferred.

**VS Code extension or command-line**
pyright Command line version 1.1.119

**Additional context**

We use this in `pandas/core/dtypes/base.py` and I'm getting asked why declaring the type of the variable `registry` is necessary.

**Comment by erictraut** [comment]
Is the global `registry` variable intended to be part of the official interface contract of pandas? If so, it needs a type annotation. This is required by the rules set forth in [this document](https://github.com/microsoft/pyright/blob/master/docs/typed-libraries.md).

Type inference rules are not prescribed or specified and are, in practice, inconsistent between type checkers. By providing an explicit type annotation for variables, you can eliminate any ambiguity about the interface contract.

As an added benefit, an annotation gives you an opportunity to specify additional information in cases where the variable is intended to be immutable (i.e. `registry: Final[Registry] = Registry()`.

**Comment by Dr-Irv** [comment]
It's not meant to be public in pandas, so we'll fix it that way.  Thanks.
